@xen-orchestra/disk-transform/src/Disk.mts

```diff
@@ -0,0 +1,93 @@
+export type DiskBlockData = Buffer
+export type DiskBlock = {
+  index: number // the index of the block. Offset in raw disk is index * blockSize
+  data: DiskBlockData // thue Buffer like data of this block. Must be blockSize length
+}
+
+export type BytesLength = number
+
+export abstract class Disk {
+  #generatedDiskBlocks = 0 
+  #parent?: Disk
+  get parent(): Disk | undefined {
+    return this.#parent
+  }
+
+  abstract getVirtualSize(): number
+  abstract getBlockSize(): number
+  abstract init(): Promise<void>
+  abstract close(): Promise<void>
+
+  abstract isDifferencing(): boolean
+  // optional method
+  instantiateParent(): Disk {
+    throw new Error('Method not implemented.')
+  }
+  async openParent(): Promise<Disk> {
```

Comment:
Will it be much change to split your class?
Having a class `DiskWithParent` and `DiskWithoutParent` for example?
Because, i don't like that your class exposes  `openParent` methods available for all `child class` but will not be usable for some of them.